src/dbus_mpris.rs

```diff
@@ -281,23 +281,19 @@ async fn create_dbus_server(
                 if playback.device.name == mv_device_name {
                     let new_pos = playback
                         .progress
-                        .and_then(|d| d.as_millis().try_into().ok())
-                        .and_then(|d: i64| d.checked_add(pos / 1000));
+                        .and_then(|d| d.checked_add(&Duration::milliseconds(pos / 1000)));
 
                     if let Some(new_pos) = new_pos {
-                        let duration: u32 = match playback.item {
-                            Some(PlayableItem::Track(t)) => t.duration.as_millis(),
-                            Some(PlayableItem::Episode(e)) => e.duration.as_millis(),
+                        let duration: Duration = match playback.item {
+                            Some(PlayableItem::Track(t)) => t.duration,
+                            Some(PlayableItem::Episode(e)) => e.duration,
                             None => return Ok(()),
-                        }
-                        .try_into()
-                        .unwrap_or(u32::MAX);
+                        };
 
                         // MPRIS spec: negative values should be treated as 0
-                        let new_pos = new_pos.max(0);
-                        if new_pos <= duration as i64 {
-                            let _ =
-                                sp_client.seek_track(new_pos as u32, playback.device.id.as_deref());
+                        let new_pos = new_pos.max(Duration::zero());
```

Comment:
Is this line necessary, now that we're using Durations? I'd assume that Durations could never be negative by the type itself?

Reply:
std::Duration only allows positive values, but chrono::Duration allows negative and is what rspotify uses for everything. So I think it is still useful to make sure `new_pos` is >= 0? But I could be wrong.

Reply:
Oh, yeah. It's actually quite important that they support negative values for us to be able to seek backwards. 😅